tests/e2e/deployment/deployment_test.go

```diff
@@ -194,4 +207,105 @@ var _ = Describe("Application deployment test in E2E scenario", func() {
+			// Start test timer
+			testTimer = DeploymentTestTimerGroup.NewTestTimer(testSpecReport.LeafNodeText)
+		})
+
+		AfterEach(func() {
+			// End test timer
+			testTimer.End()
+			// Print result
+			testTimer.PrintResult()
+
+			By(fmt.Sprintf("get StatefulSet %s", UID))
+			deployment, err := utils.GetStatefulSet(clientSet, metav1.NamespaceDefault, UID)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("list pod for StatefulSet %s", UID))
+			labelSelector := labels.SelectorFromSet(map[string]string{"app": UID})
+			_, err = utils.ListPods(clientSet, metav1.NamespaceDefault, labelSelector, nil)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("delete StatefulSet %s", UID))
+			err = utils.DeleteStatefulSet(clientSet, deployment.Namespace, deployment.Name)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("wait for pod of StatefulSet %s disappear", UID))
+			err = utils.WaitForPodsToDisappear(clientSet, metav1.NamespaceDefault, labelSelector, constants.Interval, constants.Timeout)
+			Expect(err).To(BeNil())
+
+			utils.PrintTestcaseNameandStatus()
+		})
+
+		It("Basic StatefulSet test", func() {
+			replica := int32(2)
+			// Generate the random string and assign as a UID
+			UID = "edge-statefulset-" + utils.GetRandomString(5)
+
+			By(fmt.Sprintf("create deployment %s", UID))
+			d := utils.NewTestStatefulSet(UID, ctx.Cfg.AppImageURL[1], replica)
+			_, err := utils.CreateStatefulSet(clientSet, d)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("get deployment %s", UID))
+			_, err = utils.GetStatefulSet(clientSet, corev1.NamespaceDefault, UID)
+			Expect(err).To(BeNil())
+
+			time.Sleep(time.Second * 1)
+
+			By(fmt.Sprintf("get pod for deployment %s", UID))
+			labelSelector := labels.SelectorFromSet(map[string]string{"app": UID})
+			podList, err := utils.ListPods(clientSet, corev1.NamespaceDefault, labelSelector, nil)
+			Expect(err).To(BeNil())
+			Expect(podList).NotTo(BeNil())
+
+			By(fmt.Sprintf("wait for pod of deployment %s running", UID))
+			utils.WaitforPodsRunning(ctx.Cfg.KubeConfigPath, *podList, 240*time.Second)
+		})
+
+		It("Delete statefulSet pod multi times", func() {
+			replica := int32(2)
+			// Generate the random string and assign as a UID
+			UID = "edge-statefulset-" + utils.GetRandomString(5)
+
+			By(fmt.Sprintf("create deployment %s", UID))
+			d := utils.NewTestStatefulSet(UID, ctx.Cfg.AppImageURL[1], replica)
+			_, err := utils.CreateStatefulSet(clientSet, d)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("get pod for deployment %s", UID))
+			labelSelector := labels.SelectorFromSet(map[string]string{"app": UID})
+			podList, err := utils.ListPods(clientSet, corev1.NamespaceDefault, labelSelector, nil)
+			Expect(err).To(BeNil())
+			Expect(podList).NotTo(BeNil())
+
+			By(fmt.Sprintf("wait for pod of deployment %s running", UID))
+			utils.WaitforPodsRunning(ctx.Cfg.KubeConfigPath, *podList, 240*time.Second)
+
+			deletePodName := fmt.Sprintf("%s-1", UID)
+			for i := 0; i < 5; i++ {
```

Comment:
is this right? create statefulset only once, but delete pod 5 times?

Reply:
for statefulet, we delete the pod, the kube-controller will create it immedally, so I wait the pod running.
Without the fix, if we delete the statefulset pod, the new pod will be pending and will not run again